### ISSUE TITLE
web2 client is confused about stream/consumer API

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3104]
|<img alt="htj's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| htj reported|
|-|-|
|Trac ID|trac#3104|
|Type|defect|
|Created|2008-03-20 08:55:57Z|

The web2 client appears to be using the consumer API instead of the stream API on web2.client.http in line 214. I think it should be close() instead of finish().

Have attached patch (untested, sorry).

Attachments:

* [web2_client_stream_api_fix.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/8e3/8e338d1c36d173dba11c9a6928fe02a445b39b12/48f4238db35c37d1039cab672f13228cf0919f86.diff) (297 bytes) - added by htj on 2008-03-20 08:56:50Z - Fixes the web2 client to use the stream API on streams

<details><summary>Searchable metadata</summary>

```
trac-id__3104 3104
type__defect defect
reporter__htj htj
priority__normal normal
milestone__Web2_Gold_Master Web2-Gold-Master
branch__ 
branch_author__ 
status__closed closed
resolution__wontfix wontfix
component__web2 web2
keywords__ 
time__1206003357000000 1206003357000000
changetime__1297616775000000 1297616775000000
version__None None
owner__dreid dreid

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="htj's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|htj commented|
|-|-|

enhanchement -> defect

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set status to `closed`|
|-|-|

Not going to fix this in web2 (See [#4821](https://github.com/twisted/twisted/issues/4821)).  The new Twisted Web client API (`twisted.web.client.Agent`) has a pretty clear idea of how to stream data, so I don't think there's a corresponding problem in Twisted Web to fix, so not retargeting this at the web component.
